index.js

```diff
@@ -7,125 +7,145 @@
  * config.announce is an array of trackers to publish and check for torrents on
  *
 */
-var config = require('./config.json')
-require('electron-debug')({showDevTools: true})
+const config = require('./config.json')
+require('electron-debug')({ showDevTools: true })
 
 /* Begin Dependencies */
 
 // We use path to locate files on the local filesystem
-var path = require('path')
+const path = require('path')
 
 // Electron is used to render a browser for the user
-var electron = require('electron')
+const electron = require('electron') // eslint-disable-line import/no-unresolved
+
+// IPC is used to convey torrent information to rendering process
+const ipcMain = electron.ipcMain
 
 // WebTorrent gives our application the ability to download/seed torrents
-var WebTorrent = require('webtorrent')
+const WebTorrent = require('webtorrent')
 
 /* End Dependencies */
 
 // Create a new client responsible for seeding/downloading torrents
-var client = new WebTorrent()
-
-// registerTorrentProtocol takes an instance of electron and registers a
-// handler for our new protocol, allowing the instance of electron to resolve
-// requests against a torrent
-function registerTorrentProtocol(electron, cb) {
-  electron
-    .protocol
-    .registerFileProtocol(config.protocol, peerProtocolHandler,
-                          function registeredProtocol(e) {
-                            return cb(e)
-                            // Don't treat our new protocol like http
-                            electron.registerStandardSchemes([config.protocol])
-                            // Done setting up our new protocol
-                            return cb()
-                          })
-}
+const client = new WebTorrent()
+
+// ipc Global Status channel , only for recieving Stat requests
+// Reply sent on : global-status-reply
+ipcMain.on('global-status', function updateStatus (event, arg) {
+  // status contains WebTorrent Client Stats
+  const status = {
+    download: client.downloadSpeed,
+    upload: client.uploadSpeed,
+    torrents: client.torrents
```

Comment:
Right now I'm blindly returning the entire array. We can optimize this if it becomes a performance issue by being able to request the "full" response when the info bar slides out into a full dashboard and, otherwise, return just what the info bar needs.